my-app/src/PopUp.js

```diff
@@ -1,8 +1,79 @@
-import React, { useState } from "react";
+import { Autocomplete, useLoadScript } from "@react-google-maps/api";
+import React, { useRef, useState } from "react";
 
 import "./PopUp.css";
 
-export default function PopUp({ onClose }) {
+const libraries = ["places"];
+
+export default function PopUp({ isOpen, onClose }) {
+  const initialFormState = {
+    ProjectName: "",
+    Describe: "",
+    Location: "",
+    Image: null,
+  };
+
+  const [form, setForm] = useState(initialFormState);
+  const autocompleteRef = useRef(null);
+  const { isLoaded } = useLoadScript({
+    googleMapsApiKey: "AIzaSyCIdAofDyeBU8kc9VcVfBPGa30voIG7klc",
+    libraries,
+  });
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    try {
+      const formData = new FormData();
+
+      formData.append("ProjectName", form.ProjectName);
+      formData.append("Describe", form.Describe);
+      formData.append("Location", form.Location);
+      formData.append("Image", form.Image);
+
+      const response = await fetch("http://localhost:3000/uploads", {
+        method: "POST",
+        body: formData,
+      });
+
+      if (!response.ok) {
+        throw new Error("Failed to save the project data.");
+      }
+
+      onClose();
+    } catch (error) {
+      console.error("Error:", error);
+    }
+  };
+
+  const handlePlaceChanged = () => {
+    const place = autocompleteRef.current.getPlace();
+    if (place?.geometry) {
+      const {
+        formatted_address,
+        geometry: { location: loc },
+      } = place;
+      const { lat, lng } = loc;
+      setForm((prevForm) => ({
+        ...prevForm,
+        Location: formatted_address,
+        Latitude: lat(),
```

Comment:
Quick question, why is the value set as `lat()` instead of just `lat`?

Reply:
Sorry for the confusion here. In this code I think I was trying to get the lng and lat so I could store that in my database but it wasnt used because I ended up doing this an entirely different way. I basically got their addresses and use geocode to get the lng and lat instead